driver-kotlin-extensions/src/test/kotlin/com/mongodb/kotlin/client/model/AggregatesTest.kt

```diff
@@ -0,0 +1,357 @@
+                }} """,
+            densify(Customer::email, DensifyRange.fullRangeWithStep(1)))
+
+        assertEquals(
+            """  {"${'$'}densify": {
+                    "field": "email",
+                    "range": { "bounds": "full", "step": 1 },
+                    "partitionByFields": ["foo"]
+                }} """,
+            densify(
+                Customer::email,
+                range = DensifyRange.fullRangeWithStep(1),
+                options = DensifyOptions.densifyOptions().partitionByFields("foo")))
+    }
+
+    @Test
+    @Suppress("LongMethod")
+    fun accumulators() {
+        assertEquals(com.mongodb.client.model.Accumulators.sum("age", 1), sum(Person::age, 1))
+
+        assertEquals(com.mongodb.client.model.Accumulators.avg("age", 1), avg(Person::age, 1))
+
+        assertEquals(
+            com.mongodb.client.model.Accumulators.percentile("age", 1, 2, QuantileMethod.approximate()),
+            percentile(Person::age, 1, 2, QuantileMethod.approximate()))
+
+        assertEquals(
+            com.mongodb.client.model.Accumulators.median("age", 1, QuantileMethod.approximate()),
+            median(Person::age, 1, QuantileMethod.approximate()))
+
+        assertEquals(com.mongodb.client.model.Accumulators.first("age", 1), first(Person::age, 1))
+
+        assertEquals(com.mongodb.client.model.Accumulators.firstN("age", 1, 2), firstN(Person::age, 1, 2))
+
+        assertEquals(
+            com.mongodb.client.model.Accumulators.top("age", com.mongodb.client.model.Sorts.ascending("name"), 1),
+            top(Person::age, ascending(Person::name), 1))
+
+        assertEquals(
+            com.mongodb.client.model.Accumulators.topN("age", com.mongodb.client.model.Sorts.ascending("name"), 1, 2),
+            topN(Person::age, ascending(Person::name), 1, 2))
+
+        assertEquals(com.mongodb.client.model.Accumulators.last("age", 1), last(Person::age, 1))
+
+        assertEquals(com.mongodb.client.model.Accumulators.lastN("age", 1, 2), lastN(Person::age, 1, 2))
+
+        assertEquals(
+            com.mongodb.client.model.Accumulators.bottom("age", com.mongodb.client.model.Sorts.ascending("name"), 1),
+            bottom(Person::age, ascending(Person::name), 1))
+
+        assertEquals(
+            com.mongodb.client.model.Accumulators.bottomN(
+                "age", com.mongodb.client.model.Sorts.ascending("name"), 1, 2),
+            bottomN(Person::age, ascending(Person::name), 1, 2))
+
+        assertEquals(com.mongodb.client.model.Accumulators.max("age", 1), max(Person::age, 1))
+
+        assertEquals(com.mongodb.client.model.Accumulators.maxN("age", 1, 2), maxN(Person::age, 1, 2))
+
+        assertEquals(com.mongodb.client.model.Accumulators.min("age", 1), min(Person::age, 1))
+
+        assertEquals(com.mongodb.client.model.Accumulators.minN("age", 1, 2), minN(Person::age, 1, 2))
+
+        assertEquals(com.mongodb.client.model.Accumulators.push("age", 1), push(Person::age, 1))
```

Comment:
It is too bad there is no simple way to automate this, and ensure that all methods, including future methods, are tested, similarly to how the ExtensionsApiTest uses reflection to confirm that all methods are implemented.

In NotNullApiTest, I used reflection to automate something similar, but this may not be as straightforward here. Just something to consider.

Reply:
Good point 👍 I think the project needs a more holistic approach like adding [Kover](https://www.youtube.com/watch?v=jNu5LY9HIbw&ab_channel=KotlinbyJetBrains) which is a Kotlin specific code coverage that can produce JaCoCo compatible report. I added https://jira.mongodb.org/browse/JAVA-5705

Reply:
Nice.